python/servo/testing_commands.py

```diff
@@ -146,13 +149,35 @@ def test_unit(self, test_name=None, package=None):
         self.ensure_bootstrapped()
 
         if package:
-            packages = [package]
+            packages = {package}
         else:
-            packages = os.listdir(path.join(self.context.topdir, "tests", "unit"))
+            packages = set()
+
+        test_patterns = []
+        for test in test_name:
+            # add package if 'tests/unit/<package>'
+            match = re.search("tests/unit/(\\w+)/?$", test)
```

Comment:
Nevermind. I don't think there's anything in `fnmatch` that allows for capturing of text within filepaths like you're doing here